### PR TITLE
fix: apply fix for search being submit-able 

### PR DESCRIPTION
**Pull request by petesfrench**
## Done

- apply fix for search being submit-able 

### Flyby work

(I have added comments pointing out the changes as there are a lot of files)
- Format js
- Add `js-submit-button` where is was missing to enable the loading spinner
- Add functionality so that if you click outside the modal it closes 

## QA

- Open [demo page with a form](https://canonical-com-1431.demos.haus/solutions/ai)
- Open the the modal, click inside the modal and see nothing happens. Click outside the modal and see that it closes.
- Open the modal, fill it in and submit it. See that the submit button is disabled and have the [vanilla loading spinner](https://vanillaframework.io/docs/patterns/icons)
- Check this for a few of the different forms this change was applied to

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-16074

### CODE REVIEW

File: static/js/forms.js [hunk 1]
```diff
@@ -92,38 +92,41 @@ function requiredCheckbox(fieldset, target) {
...
 
-const form = document.querySelector("form");
-const submitButton = form.querySelector('button[type="submit"]');
-if(requiredFieldset) submitButton.disabled = true;
+  const submitButton = form.querySelector('button[type="submit"]');
+  const requiredFieldset = form.querySelectorAll(
+    "fieldset.js-required-checkbox",
+  );
+  // By default we disable the button, until the required fields are selected
+  if (requiredFieldset.length) submitButton.disabled = true;
```
Comment (petesfrench) [inline static/js/forms.js:111]: This is the fix for the search

File: static/js/modals.js [hunk 2]
```diff
@@ -84,6 +84,13 @@
     return false;
   }
 
+  // Handle click outside modal
```
Comment (petesfrench) [inline static/js/modals.js:87]: Here we add the functionality to close modal when clicking outside

**Review by britneywwc — COMMENTED** [review]
The search bar and JS form modal functions as expected. I left a few minor comments about some bugs found while QA-ing the forms

File: templates/careers/application/_withdrawal-form.html [hunk 3]
```diff
@@ -24,7 +24,7 @@ <h2 class="u-no-margin--bottom" id="modal-title">Application withdrawal</h2>
           </div>
           <footer class="p-withdrawal-modal__footer">
             <button class="p-button--base" aria-controls="withdrawal-modal">Cancel</button>
-            <button type="submit" name="submit" class="p-button--positive">Send email</button>
+            <button type="submit" name="submit" class="p-button--positive js-submit-button">Send email</button>
```
[the remark this answers is not in the document]
Comment (petesfrench) [reply templates/careers/application/_withdrawal-form.html:27]: Let's remove it from here then

File: templates/careers/application/partial/_assessment.html [hunk 4]
```diff
@@ -47,7 +47,7 @@ <h3 class="p-heading--2 p-process-stepped-list__title--stepper-{{ assessment_sta
                       aria-controls="request-assessment-modal">Request assessment report</button>
             </p>
             <div class="p-modal" id="request-assessment-modal">
-              <section class="p-modal__dialog"
+              <section class="p-modal__dialog js-modal-content"
```
[the remark this answers is not in the document]
Comment (petesfrench) [reply templates/careers/application/partial/_assessment.html:50]: Same here

File: templates/microcloud/_microcloud-modal.html [hunk 5]
```diff
@@ -150,7 +150,7 @@ <h2 class="p-heading--1">Let's get in touch</h2>
               <div class="pagination">
                 <hr class="p-rule--muted" />
                 <button type="submit"
-                        class="pagination__link--next p-button--positive"
+                        class="pagination__link--next p-button--positive js-submit-button"
```
[the remark this answers is not in the document]
Comment (petesfrench) [reply templates/microcloud/_microcloud-modal.html:153]: Nice catch! I will also add the required class to the form fields

File: templates/blog/newsletter-form.html [hunk 6]
```diff
@@ -14,7 +14,7 @@
         href="https://www.ubuntu.com/legal/dataprivacy" target="_blank">Canonical's Privacy Policy</a>.</p>
   </div>
 
-  <button type="submit" class="u-no-margin--bottom">Sign up</button>
+  <button type="submit" class="u-no-margin--bottom js-submit-button">Sign up</button>
```
[the remark this answers is not in the document]
Comment (petesfrench) [reply templates/blog/newsletter-form.html:17]: Removing

**Comment by petesfrench** [comment]
@britneywwc Thanks for the review! Can you take another look pls

**Review by britneywwc — APPROVED** [review]
thanks for the changes, looks great! 🧊

**Pull request merged**